AE-1992 Add tool to generate a large number of energiatodistus

muep opened the pull request:
It is not necessarily a very polished one, but works for populating a large data set into the local development database.

muep committed:
AE-1992 Add tool to generate a large number of energiatodistus

It is not necessarily a very polished one, but works for populating a
large data set into the local development database.

solita-antti-mottonen commented:
Lisäätkö maininnan johonkin readmehen että tämmöinen on ja miten sitä ajetaan?

solita-juhohaa committed:
AE-1992: Add logging to performance testing generator function

solita-juhohaa committed:
AE-1992: Add instructions on generating energiatodistukset

Juholei commented on etp-backend/src/dev/clj/solita/etp/energiatodistus_generator.clj:
```diff
@@ -0,0 +1,42 @@
+;; This is perhaps not 100% accurate representation on how production data
+;; looks, but it should be close enough for performance testing purposes.
+;; The added energiatodistukset are just set to signed directly.
+(defn add-energiatodistukset! [db laatija-id versio energiatodistukset]
```
Tämän kommentin voisi siirtää docstringiksi funktiolle

Juholei commented on etp-backend/src/dev/clj/solita/etp/energiatodistus_generator.clj:
```diff
@@ -0,0 +1,42 @@
+      (map vals db-rows)
+      db/default-opts)))
+
+;; Will also add a laatija for the added energiatodistukset.
```
Tähänkin voisi kokonaisuutena selventää mitä funktio tekee ja siirtää docstringiksi.

Juholei commented on etp-backend/src/dev/clj/user.clj:
```diff
@@ -74,3 +74,8 @@
        (walk/postwalk
          #(if (xschema/map-literal? %) (into (sorted-map) %) %))))
+
+(defn generate-energiatodistukset [count]
```
Tällekin docstring ja ! nimeen

Juholei commented on etp-backend/src/dev/clj/solita/etp/energiatodistus_generator.clj:
```diff
@@ -0,0 +1,42 @@
+      db/default-opts)))
+
+;; Will also add a laatija for the added energiatodistukset.
+(defn add-energiatodistukset-for-performance-testing [db n]
```
Nimeen ! kertomaan että tekee sivuvaikutuksia

solita-juhohaa committed:
AE-1992: Document the functions better

Also changed the implementation to not drop out counts less than
the batch-size. This simplifies the documentation also.

Juholei commented on etp-backend/src/dev/clj/solita/etp/energiatodistus_generator.clj:
```diff
@@ -7,14 +7,15 @@
-;; looks, but it should be close enough for performance testing purposes.
-;; The added energiatodistukset are just set to signed directly.
-(defn add-energiatodistukset! [db laatija-id versio energiatodistukset]
+(defn add-and-shortcut-signing-energiatodistukset-for-performance-testing! [db laatija-id versio energiatodistukset]
```
Tässä on nyt parametrilista ja docstring väärin päin

solita-juhohaa committed:
AE-1992: Simplify the responsibilities of the energiatodistus generating functions

solita-juhohaa commented on etp-backend/src/dev/clj/solita/etp/energiatodistus_generator.clj:
```diff
@@ -0,0 +1,48 @@
+                     :allekirjoitusaika (Instant/now)
+                     ;; Set voimassaolo-paattymisaika 10 years into the future.
+                     :voimassaolo-paattymisaika (-> (Instant/now)
+                                                    (.plusSeconds 315360000)))
```
Tähän olisi voinut kommentoida, että nämä ovat juuri ne sarakkeet, jotka tulee asettaa, jotta energiatodistus on allekirjoitettu, mutta en ole varma onko se niin, vai onko tässä jotain muutakin.

muep replied:
Täydensin tähän hieman

solita-juhohaa commented on README.md:
````diff
@@ -124,6 +124,19 @@ both databases with a single call.
+```clojure
+(user/generate-energiatodistukset-for-performance-testing! 2000)
+```
+
````
Tähän voisi lisätä esimerkin siitä kuinka voi ajaa komennon, jolla ladataan csv.

solita-juhohaa replied:
Tajusin tässä, että senhän saa esim. chrome dev toolseista, kun käy https://localhost:3000/#/energiatodistus/all lataamassa [Lataa energiatodistukset CSV-tiedostona](https://localhost:3000/api/private/energiatodistukset/csv/energiatodistukset.csv?order=desc&sort=energiatodistus.id)

![image](https://github.com/solita/etp-core/assets/143187061/a3a8bb5a-2e38-4477-8846-9030b79573cb)

muep committed:
AE-1992 Some cleanup in the mock signing
